CHANGELOG.md

```diff
@@ -30,6 +38,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
   logging so that it can be imported directly by subprojects without
   needing to worry about wierd output or without the log handing getting
   in the way of your project.
+- Linting errors in the final file are now reported with their position
+  in the source file rather than in the templated file. This means
```

Comment:
Would it be useful to report both the source file and target file lines? I imagine for big macros you would want the possibility to look at the compiled file (I do this today)

Reply:
Potentially yes, I totally see your point. I'm not quite sure _how_ though. From a data structure point of view, the source and templated positions are both accessible within the new enriched position markers. We're just choosing to display one over the other. I briefly considered making it something configurable - would you make use of that feature if I made it a flag in the config file?

If running in a more verbose mode, you'll also already get an output of the compiled file in the logs if you want it for debugging, but I'm concious that you *also* get a lot of other logging. In big files it's a very very long log.

Totally up for suggestions on how to make this more useful!

Reply:
Do you think it would be feasible to compare the Line number and position of the source vs templated file and if they differ add a column in the output?

Like:
```
L:   3 | P:   5 | TL: 7 | TP: 10 | L027 | Unqualified reference 'current_timestampp' found in
                        | select with more than one referenced table/view.
```

Where TL and TP stand for Templated Line and Templated Position

Reply:
I like the TL / TP idea! Or something along those lines.

This is kind of "off topic", but at my job, we use the `pygments` package to output SQL errors (parse errors, etc.) with color syntax highlighting. I wonder if something similar could be a good feature for SQLFluff at some point. It has helped a bit with being able to quickly read and interpret error output.

Reply:
I like both of these ideas, but I think both need fleshing out and we should make seperate issues for them.

@barrywhart - could you make one for the colour coded errors idea. In particular what you think should determine the colours?

@dmateusp - could you make another one for extending the position output to something with more detail in it? In my mind I'm not sure how someone would *use* the templated position, because that's not what they're actually looking at in an editor - but I think there's definitely something in there, because if users look closely at the moment there are *some* elements where I can't conclusively identify their source position and so display the templated position only, and that will likely confuse users at some point.

Reply:
Not sure I understand the first part of your question ^^', but after thinking more about this, I think it deserves some experimentation and focused discussion, worthy of its own PR

Reply:
Sure, I'll create a ticket. As a first approximation, it'd be "do what Pygments does now, but using the SQLFluff parser" (since it has more support for the carrots dialects of SQL).